src/middlewared/middlewared/utils/osc/linux/user_context.py

```diff
@@ -1,25 +1,94 @@
 # -*- coding=utf-8 -*-
 import logging
+import os
+import pwd
+import queue
 import subprocess
 
+from multiprocessing import Process, Queue, Value
+from typing import Callable, Optional
+
 logger = logging.getLogger(__name__)
 
-__all__ = ["run_command_with_user_context"]
+__all__ = ['run_command_with_user_context', 'set_user_context']
+
+
+def set_user_context(user: str) -> None:
+    user_details = pwd.getpwnam(user)
+    os.setgroups(os.getgrouplist(user, user_details.pw_gid))
+    os.setresgid(user_details.pw_gid, user_details.pw_gid, user_details.pw_gid)
+    os.setresuid(user_details.pw_uid, user_details.pw_uid, user_details.pw_uid)
+
+    if any(
+        c() != v for c, v in (
+            (os.getuid, user_details.pw_uid),
+            (os.geteuid, user_details.pw_uid),
+            (os.getgid, user_details.pw_gid),
+            (os.getegid, user_details.pw_gid),
+        )
+    ):
+        raise Exception(f'Unable to set user context to {user!r} user')
+
+    try:
+        os.chdir(user_details.pw_dir)
+    except Exception:
+        os.chdir('/')
 
```

Comment:
I don't believe cwd should be relevant for this, so perhaps chdir() into /var/empty?